src/components/data-display/tables/ReturnedNamesTable.tsx

```diff
@@ -0,0 +1,484 @@
+                  >
+                    {formatForMaxCharCount(decodeDomainToASCII(rowValue), 20)}{' '}
+                    <ExternalLinkIcon
+                      width={'12px'}
+                      height={'12px'}
+                      fill={'var(--text-white)'}
+                    />
+                  </Link>
+                }
+              />
+            );
+          }
+          case 'closingDate': {
+            return formatDate(rowValue);
+          }
+          case 'initiator': {
+            return isArweaveTransactionID(rowValue) ? (
+              <ArweaveID
+                id={new ArweaveTransactionID(rowValue)}
+                shouldLink
+                characterCount={12}
+                type={ArweaveIdTypes.ADDRESS}
+              />
+            ) : (
+              rowValue
+            );
+          }
+          case 'leasePrice':
+          case 'permabuy': {
+            if (rowValue instanceof Error)
+              return (
+                <Tooltip
+                  message={rowValue.message}
+                  icon={
+                    <span className="text-error">
+                      Price Error{' '}
+                      <CircleAlertIcon width={'18px'} height={'18px'} />
+                    </span>
+                  }
+                />
+              );
+            if (rowValue < 0)
+              return (
+                <span className="text-white animate-pulse">Loading...</span>
+              );
+            return `${formatARIOWithCommas(
+              new mARIOToken(rowValue).toARIO().valueOf(),
+            )} ${arioTicker}`;
+          }
+          case 'returnType': {
+            return rowValue;
+          }
+
+          case 'action': {
+            return (
+              <div className="flex justify-end w-full ">
+                <span className="flex  pr-3 w-fit h-fit gap-3 items-center justify-center overflow-visible max-h-[15px]">
+                  {row.getIsExpanded() ? (
+                    <button
+                      className="p-2 py-[0.4rem] rounded text-white bg-[#303038]"
+                      onClick={() => row.toggleExpanded()}
+                    >
+                      Close
+                    </button>
+                  ) : (
+                    <button
+                      className="p-2 py-[0.4rem] rounded text-grey border border-dark-grey hover:border-grey hover:text-white whitespace-nowrap"
+                      onClick={() => row.toggleExpanded()}
+                    >
+                      View Chart
+                    </button>
+                  )}
+                  <button
+                    className="p-2 py-[0.4rem] text-center size-fit rounded text-black bg-primary"
+                    onClick={() => {
+                      navigate(
+                        `/register/${lowerCaseDomain(row.original.name)}`,
+                      );
+                    }}
+                  >
+                    Buy
```

Comment:
Odd this is Buy even if you're going to lease. Seems like this should be "Purchase" or "Register" (It's "Register" on the main page with new names.)

Reply:
Register would be better, yeah

Reply:
https://github.com/ar-io/arns-react/pull/639/commits/fde86082de9ce23903f841989728d0751f3ecb7f